worker_test.go

```diff
@@ -45,75 +60,79 @@ func TestMain(m *testing.M) {
 		panic(err)
 	}
 
+	target, err = ioutil.TempDir("", "mistry-tests-results")
+	if err != nil {
+		panic(err)
+	}
+
 	result := m.Run()
 
 	err = os.RemoveAll(cfg.BuildPath)
 	if err != nil {
 		panic(err)
 	}
+	err = os.RemoveAll(target)
+	if err != nil {
+		panic(err)
+	}
 
 	os.Exit(result)
 }
 
 func TestSimpleBuild(t *testing.T) {
-	jr := JobRequest{Project: "simple", Params: params, Group: ""}
-	res, err := postJob(jr)
+	cmdOut, err := build("--project", "simple")
 	if err != nil {
-		t.Fatal(err)
+		t.Fatalf("Error output: %s, err: %v", cmdOut, err)
 	}
-	assert(res.ExitCode, 0, t)
 }
 
 func TestUnknownProject(t *testing.T) {
-	t.Skip("Enable when we convert tests to end-to-end")
-	jr := JobRequest{Project: "idontexist", Params: params, Group: ""}
-	res, err := postJob(jr)
-	if err != nil {
-		panic(err)
+	expected := "Unknown project"
+
+	cmdOut, err := build("--project", "Idontexist")
+	if !strings.Contains(cmdOut, expected) {
+		t.Fatalf("Error output: %s, actual: %v, expected: %v.", cmdOut, err.Error(), expected)
 	}
-	fmt.Printf("%#v\n", res)
 }
 
 func TestBuildCoalescing(t *testing.T) {
-	var result1, result2 *BuildResult
 	var wg sync.WaitGroup
 
-	jr := JobRequest{"build-coalescing", params, "foo"}
-
 	wg.Add(1)
 	go func() {
 		defer wg.Done()
-		var err error
-		result1, err = postJob(jr)
+		cmdOut, err := build("--project", "build-coalescing", "--group", "foo")
 		if err != nil {
-			log.Fatal(err)
+			t.Fatalf("Error output: %s, err: %v", cmdOut, err)
```

Comment:
Only the main goroutine should call `t.Fatal` per https://golang.org/pkg/testing/#T.FailNow.

This should remain as it was.

Reply:
Fixed, thanks for clarifying this.